frontend/ui/widget/numberpickerwidget.lua

```diff
@@ -146,7 +146,51 @@ function NumberPickerWidget:init()
                             text = _("OK"),
                             is_enter_default = true,
                             callback = function()
-                                local input_value = tonumber(input_dialog:getInputText())
+                                local input_text = input_dialog:getInputText()
+                                local input_value = tonumber(input_text)
+                                local turn_off_checks = false
+
+                                -- if the first character of the input string is ":" turn of min-max checks
+                                if input_text:match("^:") and G_reader_settings:isTrue("debug_verbose") then
+                                    turn_off_checks = true -- turn off checks
+                                    input_text = input_text:gsub("^:", "") -- shorten input
+                                    input_value = tonumber(input_text) -- try to get value
+                                end
+
+                                -- if the input text starts with `=` try to evaluate the expression
+                                -- only allow `math.*` and no other functions to be safe here.
+                                if input_text:match("^=") then
+                                    local function evaluate_string(code)
+                                        -- only execute if there a no chars (except math.xxx) and {[]} in the user input
+                                        local check_code = code:gsub("math%.%a*", "")
+                                        if check_code:find("[%a%[%]%{%}]") then
+                                            return nil
+                                        end
+                                        code = code:gsub("^=", "return ")
+                                        local env = {math = math} -- restrict to only math functions
+                                        local func, dummy = load(code, "user_sandbox", nil, env)
```

Comment:
what is this `"user_sandbox"` ? A hardcoded string known to Lua or load() ? Or some random string ?